### DISCUSSION TITLE
How to parse markdown docs to detect code spans?

### DISCUSSION THREAD

**Discussion by tamalsaha** [yuin/goldmark#193] [General]
I would like to write a tool that scans markdown files to detect the code span (the 3 backtick sections) and extract those texts and their line number in the original file. Is something like this possible with this library? Do you have any example how to access the AST and do something like this?

Thanks.

**Comment by ghost** [comment]
I'd like to do the same sort of thing with words between tilde symbols (so a marked up document could be processed in order to remove any words identified as strikethrough words, between the tilde symbols.)

**Comment by zzwx** [comment]
All extensions are actually good examples, just have a look at how they are organized and play from there.

https://github.com/yuin/goldmark/blob/master/extension/strikethrough.go

**Reply by ghost** [reply]
Thank you! Good suggestion :) Appreciate this - I'll give it a try

**Reply by tamalsaha** [reply]
@zzwx , is there a way to detect line number in the original doc from the parser? I don't see a way.

**Reply by zzwx** [reply]
Just theorizing:
`reader.PeekLine()` seems to return a segment. May be you can use that to traverse back to original source and count the amount of new lines up to Start or Stop, depending on your needs.
Also nodes that are of `*BaseBlock` type and derivatives have `Lines()` method to refer back to the source. Others have a Segment property.

**Comment by tamalsaha** [comment]
Thanks, @zzwx ! I was able to get this to work.
